backend/src/apiserver/server/experiment_server_test.go

```diff
@@ -791,294 +792,293 @@ func TestListExperiments_Multiuser(t *testing.T) {
-				Name:        "exp1",
-				Description: "first experiment",
-				ResourceReferences: []*apiV1beta1.ResourceReference{
-					{
-						Key:          &apiV1beta1.ResourceKey{Type: apiV1beta1.ResourceType_NAMESPACE, Id: "ns1"},
-						Relationship: apiV1beta1.Relationship_OWNER,
-					},
-					{
-						Key:          &apiV1beta1.ResourceKey{Type: apiV1beta1.ResourceType_NAMESPACE, Id: "ns2"},
-						Relationship: apiV1beta1.Relationship_OWNER,
-					},
-				},
-			},
-			true,
-			"Invalid resource references for experiment.",
-		},
-		{
-			"Invalid resource type",
-			&apiV1beta1.Experiment{
-				Name:        "exp1",
-				Description: "first experiment",
-				ResourceReferences: []*apiV1beta1.ResourceReference{
-					{
-						Key:          &apiV1beta1.ResourceKey{Type: apiV1beta1.ResourceType_EXPERIMENT, Id: "exp2"},
-						Relationship: apiV1beta1.Relationship_OWNER,
-					},
-				},
-			},
-			true,
-			"Invalid resource references for experiment.",
-		},
-	}
-
-	for _, tc := range tests {
-		err := ValidateCreateExperimentRequestV1(&apiV1beta1.CreateExperimentRequest{Experiment: tc.experiment})
-		if !tc.wantError && err != nil {
-			t.Errorf("TestValidateCreateExperimentRequestV1(%v) expect no error but got %v", tc.name, err)
-		}
-		if tc.wantError {
-			if err == nil {
-				t.Errorf("TestValidateCreateExperimentRequestV1(%v) expect error but got nil", tc.name)
-			} else if !strings.Contains(err.Error(), tc.errorMessage) {
-				t.Errorf("TestValidateCreateExperimentRequestV1(%v) expect error containing: %v, but got: %v", tc.name, tc.errorMessage, err)
-			}
-		}
-	}
-}
-
-func TestValidateCreateExperimentRequest_Multiuser(t *testing.T) {
-	viper.Set(common.MultiUserMode, "true")
-	defer viper.Set(common.MultiUserMode, "false")
-	tests := []struct {
-		name         string
-		experiment   *apiV2beta1.Experiment
-		wantError    bool
-		errorMessage string
-	}{
-		{
-			"Valid",
-			&apiV2beta1.Experiment{
-				DisplayName: "exp1",
-				Description: "first experiment",
-				Namespace:   "ns1",
-			},
-			false,
-			"",
-		},
-		{
-			"Missing namespace",
-			&apiV2beta1.Experiment{
-				DisplayName: "exp1",
-				Description: "first experiment",
-			},
-			true,
-			"In multi-user mode, experiment namespace is empty. Please specify a valid namespace.",
-		},
-	}
-
-	for _, tc := range tests {
-		err := ValidateCreateExperimentRequest(&apiV2beta1.CreateExperimentRequest{Experiment: tc.experiment})
-		if !tc.wantError && err != nil {
-			t.Errorf("TestValidateCreateExperimentRequest(%v) expect no error but got %v", tc.name, err)
-		}
-		if tc.wantError {
-			if err == nil {
-				t.Errorf("TestValidateCreateExperimentRequest(%v) expect error but got nil", tc.name)
-			} else if !strings.Contains(err.Error(), tc.errorMessage) {
-				t.Errorf("TestValidateCreateExperimentRequest(%v) expect error containing: %v, but got: %v", tc.name, tc.errorMessage, err)
-			}
-		}
-	}
-}
+// func TestValidateCreateExperimentRequestV1(t *testing.T) {
```

Comment:
keep these tests?

Reply:
Similarly to other cases, the validation now happens inside the functions. I added these test cases for `CreateExperiment` and `CreateExperimentV1`.